cmd/client-nfd/subcmd/compat/validate-node.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package compat
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2/registry"
+
+	"sigs.k8s.io/node-feature-discovery/cmd/client-nfd/subcmd/compat/options"
+	nodevalidator "sigs.k8s.io/node-feature-discovery/pkg/client-nfd/compat/node-validator"
+)
+
+var (
+	image     string
+	tags      []string
+	platform  options.PlatformOption
+	plainHTTP bool
+)
+
+var validateNodeCmd = &cobra.Command{
+	Use:   "validate-node",
+	Short: "Validate node based on image compatibility metadata",
+	Long:  "Validate node based on image compatibility metadata from the NFD artifact",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		return platform.Parse(cmd)
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+
+		ref, err := registry.ParseReference(image)
+		if err != nil {
+			return err
+		}
+
+		a := &nodevalidator.Args{
+			RegReference: &ref,
+			Tags:         tags,
+			Platform:     platform.Platform,
+			PlainHTTP:    plainHTTP,
+		}
+		nv := nodevalidator.New(nodevalidator.WithArgs(a))
+
+		out, err := nv.Execute(ctx)
+		if err != nil {
+			return err
+		}
+		printResult(out)
+
+		return nil
+	},
+}
+
+func printResult(css []*nodevalidator.CompatibilityStatus) {
+	for i, cs := range css {
+		fmt.Print(text.Colors{text.FgCyan, text.Bold}.Sprintf("Compatibility set #%d ", i+1))
+		fmt.Print(text.FgCyan.Sprintf("Weight: %d", cs.Weight))
+		if cs.Tag != "" {
+			fmt.Print(text.FgCyan.Sprintf("; Tag: %s", cs.Tag))
+		}
+		fmt.Println()
+		fmt.Println(text.FgWhite.Sprintf("Description: %s", cs.Description))
+
+		for _, r := range cs.Rules {
+			printTable(r)
+		}
+		fmt.Println()
+	}
+}
+
+func printTable(r nodevalidator.RuleStatus) {
+	t := table.NewWriter()
```

Comment:
for matchAny, (refer to the example commented in the file compatibilit-6-wind-router-rule.yaml), 

```
    - name: "nic"
      matchAny:
        - matchFeatures:
          - feature: pci.device
            matchExpressions:
              vendor: {op: In, value: ["0eee"]}
              class: {op: In, value: ["0200"]}
        - matchFeatures:
          - feature: pci.device
            matchExpressions:
              vendor: {op: In, value: ["0fff"]}
              class: {op: In, value: ["0200"]}
```


the result is printed like this:

```
┌──────────────────────────────────────────────────────────────┐
│ RULE: NIC -  FAIL                                            │
├───┬───────────────────┬─────────────────────────────┬────────┤
│ # │ FEATURE           │ EXPRESSION                  │ STATUS │
├───┼───────────────────┼─────────────────────────────┼────────┤
│ 1 │ pci.device.class  │ {op: "In", value: ["0200"]} │ FAIL   │
│ 2 │ pci.device.vendor │ {op: "In", value: ["0eee"]} │ FAIL   │
│ 3 │ pci.device.class  │ {op: "In", value: ["0200"]} │ FAIL   │
│ 4 │ pci.device.vendor │ {op: "In", value: ["0fff"]} │ FAIL   │
└───┴───────────────────┴─────────────────────────────┴────────┘
```
it's little difficult to distinguish "one of the matchAny" failed, it just listed all field result in flat

Reply:
True, it's not easy to do this. Let me figure out how to represent matchAny.

Reply:
I added `matchAny` representation by adding a separtor between the terms, like (look at the `one of available nics`):
![image](https://github.com/user-attachments/assets/ea68cdc4-6439-4ae6-abc3-9cf247249fb5)

Reply:
I've also added the backreference and matchAny features to the example.